### PR TITLE
Icu wasi package

### PR DESCRIPTION
**Pull request by mkhamoyan**
Contributes to https://github.com/dotnet/runtime/issues/65895

Use icu package for wasi-wasm for Wasi flow.

### CODE REVIEW

**Review by pavelsavara — APPROVED** [review]

**Comment by mkhamoyan** [comment]
Failures are not related

File: src/mono/mono.proj [hunk 1]
```diff
@@ -434,9 +434,8 @@
       <_MonoCFLAGS Include="-D_WASI_EMULATED_PROCESS_CLOCKS"/>
       <_MonoCFLAGS Include="-D_WASI_EMULATED_SIGNAL"/>
       <_MonoCFLAGS Include="-D_WASI_EMULATED_MMAN"/>
-      <!-- TODOWASI wasi-wasm nuget instead of browser-wasm -->
-      <_MonoCFLAGS Condition="'$(MonoWasmThreads)' == 'true'" Include="$(EscapedQuoteW)-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm-threads', 'native', 'include').Replace('\','/'))$(EscapedQuoteW)"/>
-      <_MonoCFLAGS Condition="'$(MonoWasmThreads)' != 'true'" Include="$(EscapedQuoteW)-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm', 'native', 'include').Replace('\','/'))$(EscapedQuoteW)"/>
+      <_MonoCFLAGS Condition="'$(MonoWasmThreads)' == 'true'" Include="$(EscapedQuoteW)-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'wasi-wasm-threads', 'native', 'include').Replace('\','/'))$(EscapedQuoteW)"/>
```
Comment (radical) [inline src/mono/mono.proj:437]: Why was this changed from `browser-wasm*` to `wasi-wasm*`?

Comment (radical) [reply src/mono/mono.proj:437]: Oh, this is in an itemGroup for wasi!
